src/Deposit.js

```diff
@@ -1099,4 +1103,174 @@ export default class Deposit {
+    )
+  }
+
+  /**
+   * Get the first threshold for deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit can
+   * get courtesy-called.
+   * The deposit can be courtesy called using `Deposit.notifyCourtesyCall`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 125%
+   */
+  async getUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Get the threshold for severe deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit
+   * can be liquidated.
+   * Liquidation can be initiated using `Deposit.notifyUndercollateralizedLiquidation`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 110%
+   */
+  async getSeverelyUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getSeverelyUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Notify the contract that the signers are undercollateralized,
+   * and move the deposit into a pre-liquidation state.
+   */
+  async notifyCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.notifyCourtesyCall())
+  }
+
+  /**
+   * Exit the courtesy call state.
+   * Only callable if the deposit is sufficiently collateralised.
+   */
+  async exitCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.exitCourtesyCall())
+  }
+
+  /**
+   * Notify the contract that the deposit is severely undercollateralised,
+   * and begin liquidation of the signer bonds.
+   *
+   * The bonds are auctioned in a falling-price auction. The value of
+   * the bonds can be queried using `Deposit.auctionValue`, and bids placed
+   * using `Deposit.purchaseSignerBondsAtAuction`.
+   */
+  async notifyUndercollateralizedLiquidation() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyUndercollateralizedLiquidation()
+    )
+  }
+
+  /**
+   * Purchases the signer bonds and closes the liquidation auction.
+   *
+   * Note: The caller must approve the Deposit contract to spend
+   * `lotSize` amount of TBTC.
+   */
+  async purchaseSignerBondsAtAuction() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.purchaseSignerBondsAtAuction()
+    )
+  }
+
+  /**
+   * Gets the current value of signer bonds at auction.
+   * Only callable if the deposit is in the liqudation state.
+   * @return {BN} auction value in wei.
+   */
+  async auctionValue() {
+    return toBN(await this.contract.methods.auctionValue().call())
+  }
+
+  // /--------------------- Timeout helpers -------------------------
+
+  /**
+   * Notify the contract that signing group setup has timed out.
+   * Only applicable during funding.
+   */
+  async notifySignerSetupFailure() {
```

Comment:
Definitely what's driving me bananas about all these helpers is that we're basically reimplementing what `TruffleContract` gave us for free :/

Reply:
Yep yep yep 💯 Absolutely feel this horror of verbosity.

I could've done with conducting a stronger investigation into web3 beforehand, maybe testing the waters rather than diving right in. ethers.js seems a lot easier to work with in retrospect.

Reply:
Aye, though it's also worth remembering that the whole point of our library is giving a consistent interface to our contracts, so it's natural to have to untangle some of the underlying impedance mismatches ourselves.

Reply:
> ethers.js seems a lot easier to work with in retrospect.

Can confirm that ethers.js is easier for the happy path, but when you need to do something different- eg an unsigned transaction for offline signing- the interface breaks down a bit